backend/src/schema/resolvers/notifications.js

```diff
@@ -1,14 +1,80 @@
-import { neo4jgraphql } from 'neo4j-graphql-js'
+const resourceTypes = ['Post', 'Comment']
+
+const transformReturnType = record => {
+  return {
+    ...record.get('notification').properties,
+    from: {
+      __typename: record.get('resource').labels.find(l => resourceTypes.includes(l)),
+      ...record.get('resource').properties,
+    },
+    to: {
+      ...record.get('user').properties,
+    },
+  }
+}
 
 export default {
   Query: {
-    Notification: (object, params, context, resolveInfo) => {
-      return neo4jgraphql(object, params, context, resolveInfo, false)
+    notifications: async (parent, args, context, resolveInfo) => {
+      const { user: currentUser } = context
+      const session = context.driver.session()
+      let notifications
+      let whereClause
+      let orderByClause
+      switch (args.read) {
+        case true:
+          whereClause = 'WHERE notification.read = TRUE'
+          break
+        case false:
+          whereClause = 'WHERE notification.read = FALSE'
+          break
+        default:
+          whereClause = ''
+      }
+      switch (args.orderBy) {
+        case 'createdAt_asc':
+          orderByClause = 'ORDER BY notification.createdAt ASC'
+          break
+        case 'createdAt_desc':
+          orderByClause = 'ORDER BY notification.createdAt DESC'
+          break
+        default:
+          orderByClause = ''
+      }
+
+      try {
+        const cypher = `
+        MATCH (resource)-[notification:NOTIFIED]->(user:User {id:$id})
+        ${whereClause}
+        RETURN resource, notification, user
+        ${orderByClause}
+        `
+        const result = await session.run(cypher, { id: currentUser.id })
+        notifications = await result.records.map(transformReturnType)
+      } finally {
+        session.close()
+      }
+      return notifications
     },
   },
   Mutation: {
-    UpdateNotification: (object, params, context, resolveInfo) => {
-      return neo4jgraphql(object, params, context, resolveInfo, false)
+    markAsRead: async (parent, args, context, resolveInfo) => {
+      const { user: currentUser } = context
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-08-30T08:18:37Z" title="Friday, August 30th 2019, 10:18:37 am +02:00">Aug 30, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,14 +1,80 @@
-import { neo4jgraphql } from 'neo4j-graphql-js'
+const resourceTypes = ['Post', 'Comment']
+
+const transformReturnType = record => {
+  return {
+    ...record.get('notification').properties,
+    from: {
+      __typename: record.get('resource').labels.find(l => resourceTypes.includes(l)),
+      ...record.get('resource').properties,
+    },
+    to: {
+      ...record.get('user').properties,
+    },
+  }
+}
 
 export default {
   Query: {
-    Notification: (object, params, context, resolveInfo) => {
-      return neo4jgraphql(object, params, context, resolveInfo, false)
+    notifications: async (parent, args, context, resolveInfo) => {
+      const { user } = context
+      const session = context.driver.session()
+      let notifications
+      let whereClause
+      let orderByClause
+      switch (args.read) {
+        case true:
+          whereClause = 'WHERE notification.read = TRUE'
+          break
+        case false:
+          whereClause = 'WHERE notification.read = FALSE'
+          break
+        default:
+          whereClause = ''
+      }
+      switch (args.orderBy) {
+        case 'createdAt_asc':
+          orderByClause = 'ORDER BY notification.createdAt ASC'
+          break
+        case 'createdAt_desc':
+          orderByClause = 'ORDER BY notification.createdAt DESC'
+          break
+        default:
+          orderByClause = ''
+      }
+
+      try {
+        const cypher = `
+        MATCH (resource)-[notification:NOTIFIED]->(user:User {id:$id})
+        ${whereClause}
+        RETURN resource, notification, user
+        ${orderByClause}
+        `
+        const result = await session.run(cypher, { id: user.id })
+        notifications = await result.records.map(transformReturnType)
+      } finally {
+        session.close()
+      }
+      return notifications
     },
   },
   Mutation: {
-    UpdateNotification: (object, params, context, resolveInfo) => {
-      return neo4jgraphql(object, params, context, resolveInfo, false)
+    markAsRead: async (parent, args, context, resolveInfo) => {
+      const { user } = context
```


`currentUser` would be more understandable even for me and it can not be clear to newbies here.